### PR TITLE
Feature/component-organization

### LINKED ISSUE #206: Refactor component organization
**Is your feature request related to a problem? Please describe.**
As a developer, it can be a bit frustrating and complex to work within the "flat" component hierarchy.
image

As it stands, we have assets, styles, business logic utils, and the components all within the same level of `/components`

**Describe the solution you'd like**
Refactor components such that they're grouped in their own directories with related assets and styles together.

e.g.
```
AddTapModel
|- AddTapModal.js
|- AddTapModal.module.css
Head
|- Head.js
|- Head.css
|- Head.test.js
.
.
```

**Describe alternatives you've considered**
[React's official documents offer some very general guidance on file structure for projects](https://reactjs.org/docs/faq-structure.html) which might be suitable here. From the page:
- group by features or routes e.g.
```
common/
  Avatar.js
  Avatar.css
  APIUtils.js
  APIUtils.test.js
feed/
  index.js
  Feed.js
  Feed.css
  FeedStory.js
  FeedStory.test.js
  FeedAPI.js
profile/
  index.js
  Profile.js
  ProfileHeader.js
  ProfileHeader.css
  ProfileAPI.js
```
- group  by file type e.g.
```
api/
  APIUtils.js
  APIUtils.test.js
  ProfileAPI.js
  UserAPI.js
components/
  Avatar.js
  Avatar.css
  Feed.js
  Feed.css
  FeedStory.js
  FeedStory.test.js
  Profile.js
  ProfileHeader.js
  ProfileHeader.css
```

**Additional context**
As per React's official doc's page (and my personal experience), **we don't need to overthink this** and we should avoid **deeply nested hierarchies**. 

The first piece of advice is that as long as we have _some_ organization, it's better than _none_, and probably better than too much organization.

The second piece of advice addresses that when projects become too deeply nested, it's very easy to lose sight of the overall codebase. Since we don't have separate engineering/development teams that could become experts within localities of the repo, it's my opinion we pursue structures that are more straightforward and within the realm that 1-2 developers would be able to fully grasp and understand its structure.

### PR DESCRIPTION
**Pull request by rh9891**
Created new directories for the components based on features (AddTapModal, ClosestTap, Food, FoodFilter, etc.) and reorganized the files. Made additional directories to organize the actions and helper files. Modified how files were being imported. All is working as expected.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,35 +4,34 @@ import { Switch, Route } from "react-router";
...
 // import { resizeWindow } from './actions'
 import "./App.css";
-import MapPage from "./components/MapPage";
-import Mission from "./components/pages/Mission";
-import Project from "./components/pages/Project";
-import Share from "./components/pages/Share";
-import Contribute from "./components/pages/Contribute";
-import Head from "./components/Head";
+import MapPage from "./components/MapPage/MapPage";
+import Mission from "./components/Pages/Mission";
```
Comment (nwyin) [inline src/App.js:8]: I'm getting this error when running locally: 
image

Seems like the directory is named `pages` but the import uses `Pages` -- resulting in this error. Changing it to `pages` resolves the error

**Comment by nwyin** [comment]
@rh9891 Left a comment with a bug I found. Otherwise, the changes seem to be working as intended. Thanks for the great work!

If you want, we can set up a meeting to talk about PR/code review practices and maybe loop @gcardonag to talk about things we might want to work into the Phlask review workflow.

**Review by nwyin — CHANGES_REQUESTED** [review]

**Comment by gcardonag** [comment]
This MR looks good overall! Thanks for putting in all that time to clean up our file structure. Its already way easier on the eyes locally. 😄 

Regarding the conflicts reported in src/components/Toolbar.js, it seems those are because this branch is still using the older `AddTapModal` component. The following steps will fix your branch locally in case you want to take a jab at solving Git conflicts.
1. Run `git pull`
2. Run `git checkout feature/component-organization`
3. Run `git merge develop`
4. Run `git status` to see the conflict details, there should be 3 files reported.
5. Fix the conflict in `Toolbar.js` by using the version of import that goes `import { AddResourceModal } from "../AddResourceModal";`
6. Run `git add src/components/*` to accept the final version of the conflicted files.
7. Run `git commit` to resolve the conflict and then push up the changes.

If these steps don't work or you have any questions, reach out to me and I can give you a hand or help solve the conflicts on my end once the `Pages` error is resolved.

File: src/App.js (hunk 1 above)
Comment (nwyin) [reply src/App.js:8]: Fixed in https://github.com/phlask/phlask-map/pull/222/commits/17867d7015463627225ae851af75c5d0e5970dc1

**Review by nwyin — APPROVED** [review]

**Pull request merged**